### ISSUE TITLE
Repeater fields don't behave like normal fields

### ISSUE THREAD

**Issue by trigau** [bolt/bolt#5846]
Don't exactly know if it's a real bug or a problem in documentation.

As written [in documentation](https://docs.bolt.cm/3.1/fields/repeater#example-usage-in-templates), I access repeater fields by function _get()_
My repeater is define by
`paragraphes:
            type: repeater
            label: Paragraphes
            group: Paragraphes
            fields:
                paragraphe:
                    type: html
                    height: 100px
                image:
                    type: image`

Innocently, I try to use repeater fields as I usually do for such fields when they're not included in a repeater
`{% for paragraphe in record.paragraphes %}
<section>
    <aside><img src="{{ imageinfo(paragraphe.get('image')).url }}" /></aside>
    <article>{{ paragraphe.get('paragraphe') }}</article>
</section>
{% endfor %}
`

And this is where things start getting exciting :
- image field doesn't act as usual and only returns an array 'file' => "file_path" instead of the file path 
- html field returns a string "`<p>some text</p>`" which escape chars are not escaped instead of an array containing `content=>"<p>some text>/p>"` which escape chars wouldn't appear

So for me we have 2 options :
1) Doc is exhaustive and there seems to be an odd behavior when fields are contained inside a repeater
2) Doc is not exhaustive and I have missed a point
## Details
- Relevant Bolt Version: 3.1.4

**Comment by trigau** [comment]
After some investigation, let me add some complement
Here is my content definition:
```
    fields:
        image:
            type: image
        paragraphes:
            type: repeater
                image:
                    type: image
```
If I want the same behavior as root image field
```
{{ record.image }}
```
I have to write
```
{% for paragraphe in record.paragraphes %}
{{ paragraphe.image.value.file }}
{% endfor %}
```

A I wrong ?

**Comment by SahAssar** [comment]
IIRC repeater fields are hydrated in a different way (or not at all, currently), and the whole repeater field acts as a DBAL collection (also IIRC). I think this is meant to change in the future so that they have all the things normal fields have, like text/html fields being twig markup objects and so on.

Perhaps @rossriley could shed some light on this? I worked some on this for templatefields but not enough to say for sure.

**Comment by rossriley** [comment]
Yes, It was supposed to be in the last release, but I had to hold it back because it was causing problems with presentation format not getting removed in some fields when it was written back into the database.

so it'll need a little more work to allow fields to use the `present` interface method that will do those kind of transforms when used in twig, but not affect the underlying data... and this is on my todolist for the next version.

**Comment by Maelstromeous** [comment]
Related to #6015

**Comment by anketwokings** [comment]
+1, I was trying to access the 'responsive' value of a video in a repeater, but (#5471) as already noted, this does not work yet. 
Keep up the good work, the repeater looks really good, but I could definitely use the responsive video for my clients site. :)

**Comment by SahAssar** [comment]
@anketwokings Until it's fixed you can wrap the `html` value of the video in a `<div class="responsive-video"></div>`, that's mostly what the responsive value does (well, and replace the width and height, but that can also be done in twig).

**Comment by anketwokings** [comment]
@SahAssar Should have thought of that myself. :)
👍

**Comment by bobdenotter** [comment]
Fixed in #6474. 

(at least, it's much nicer now, and probably as good as we can get it reasonably)
